internal/grpcsync/callback_serializer.go

```diff
@@ -85,11 +85,10 @@ func (cs *CallbackSerializer) run(ctx context.Context) {
 			// Do nothing here. Next iteration of the for loop will not happen,
 			// since ctx.Err() would be non-nil.
 		case callback, ok := <-cs.callbacks.Get():
-			if !ok {
-				return
+			if ok {
```

Comment:
I don't think we should need to read or check `ok` at all here.  There's no way `ok` can ever be false, because the channel is only closed later.  We'll unblock the `select` only when the context is canceled or a callback is created.